### ISSUE TITLE
Document what's going on in terms of caching

### ISSUE THREAD

**Issue by djc** [crate-ci/azure-pipelines#36]
Just trying this out on Quinn right now, it looks very helpful. It would be neat if documentation (the README) what, if anything, was going on in terms of caching.

**Comment by epage** [comment]
Good point that we should document this for people unfamiliar with Azure Pipelines.  As of right now, it doesn't support any caching, so neither do the templates we provide.

**Comment by jonhoo** [comment]
Azure has only recently gotten support for caching, and it is slowly rolling out. As @epage says, the CI scripts here do not _currently_ support caching at all :) The discussion to follow on the Azure side is https://github.com/microsoft/azure-pipelines-yaml/pull/113.

**Comment by djc** [comment]
Maybe this should then be reformulated as "Add support for caching"? I recently used it with Bitbucket pipelines and found the decreased turnaround time quite attractive.
